encord/user_client.py

```diff
@@ -253,24 +255,43 @@ def get_datasets(
         Returns:
             list of (role, dataset) pairs for datasets  matching filter conditions.
         """
-        properties_filter = self.__validate_filter(locals())
-        # a hack to be able to share validation code without too much c&p
-        data = self._querier.get_multiple(
-            DatasetWithUserRole,
-            payload={
-                "filter": properties_filter,
-                "enable_storage_api": True,
-            },
+
+        res = self._api_client.get(
+            "/datasets/list",
+            params=DatasetsWithUserRolesListParams(
+                title_eq=title_eq,
+                title_like=title_like,
+                description_eq=desc_eq,
+                description_like=desc_like,
+                created_before=parse_datetime_optional(created_before),
+                created_after=parse_datetime_optional(created_after),
+                edited_before=parse_datetime_optional(edited_before),
+                edited_after=parse_datetime_optional(edited_after),
+            ),
+            result_type=DatasetsWithUserRolesListResponse,
         )
 
-        def convert_dates(dataset):
-            dataset["created_at"] = parse_datetime(dataset["created_at"])
-            dataset["last_edited_at"] = parse_datetime(dataset["last_edited_at"])
-            return dataset
+        res_user = self._api_client.get(
+            "user/current",
+            params=None,
+            result_type=GetCurrentUserResponse,
+        )
 
         return [
-            {"dataset": DatasetInfo(**convert_dates(d.dataset)), "user_role": DatasetUserRole(d.user_role)}
-            for d in data
+            {
+                "dataset": DatasetInfo(
+                    dataset_hash=str(x.dataset_hash),
+                    user_hash=res_user.user_hash,
```

Comment:
TBH I think it's actually safe-ish to just semi-quietly drop this field. The value is not really usable for anything, and before this change I think it used to be the *creating* user's hash (so you're breaking the backwards compatibility here anyway), and we don't accept this as input to any APIs. No point in pretending this does anything.

I don't suggest that we drop the field straight away, but I think we can just set it to some silly `"the value is not provided for compliance reasons"` kind of constant value; and we can/should return the `user_email` from the BE instead.

@sergei-encord -- WDYT?

Reply:
Yes, sounds fair. 
I guess if we want to be extra safe, we can convert this field into a getter and annotate it as @deperecated, so whoever tries to use it would see a warning in the logs as well. But seems like optional in this case.